### PR TITLE
remove most of absolute urls

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/docker/docker.github.io/pull/15179 and https://github.com/docker/docker.github.io/pull/15250

Some links are broken in our lab environment because of absolute URLs being used and the `DOCS_URL` env var not being taken into account properly.

This PR removes most of the absolute URLs and use local `.md` files when available, otherwise a relative URL.

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: compose/release-notes.md [hunk 1]
```diff
@@ -2113,8 +2113,8 @@ naming scheme accordingly before upgrading.
     You don't have to use it - your existing Compose files will run on Compose
     1.6 exactly as they do today.
 
-    Check the upgrade guide for full details:
-    https://docs.docker.com/compose/compose-file#upgrading
```
Comment (crazy-max) [inline compose/release-notes.md:2117]: This URL was not valid. We could not catch it with `htmlproofer` because not solved as link.

Comment (dockertopia) [reply compose/release-notes.md:2117]: That's a line in the file from 2016, probably there were changes in `compose-file` affecting this. Did we have htmlproofer then?
You say "DOCS_URL env var not being taken into account properly.", how should we take this var into account? 
I'm unaware of this topic. cc: @usha-mandya

Comment (crazy-max) [reply compose/release-notes.md:2117]: > You say "DOCS_URL env var not being taken into account properly.", how should we take this var into account?
> I'm unaware of this topic. cc: @usha-mandya

`DOCS_URL` has been added in https://github.com/docker/docker.github.io/pull/15250 to avoid hardcoded URLs that would not match the environment (stage vs lab vs prod). See for example the source of https://docs-stage.docker.com/ and you will see some URLs with `docs.docker.com` which is not right and should be `docs-stage.docker.com`.

tldr; this is an internal enhancement that should not impact your workflow.

File: desktop/previous-versions/edge-releases-mac.md [hunk 2]
```diff
@@ -2287,7 +2287,7 @@ events or unexpected unmounts.
 
 **Bug fixes and minor changes**
 
-* Documentation moved to [https://docs.docker.com/desktop/mac/](../index.md)
+* Documentation moved to [{{ site.docs_url }}/desktop/mac/](../index.md)
```
Comment (usha-mandya) [inline desktop/previous-versions/edge-releases-mac.md:2290]: Do we need `{{ site.docs_url }}` here? Wondering if it should be the same as other relative URLs on this page?

**Comment by usha-mandya** [comment]
@crazy-max Added a minor comment on one of the Desktop pages. Looks good otherwise. Thank you!

File: desktop/previous-versions/edge-releases-mac.md (hunk 2 above)
Comment (crazy-max) [reply desktop/previous-versions/edge-releases-mac.md:2290]: I think here it's a implicitly showing the absolute url but agree that `/desktop/mac/` is enough, will fix that.

**Comment by crazy-max** [comment]
@usha-mandya Should be good, thanks for the review

File: compose/release-notes.md (hunk 1 above)
Comment (dockertopia) [reply compose/release-notes.md:2209]: Got it, so to get right base URL according with environment. Thanks for the explanation.

**Review by usha-mandya — APPROVED** [review]
LGTM. Thank you
